File: DSA/Nivedan_590014904/Day_12_Question_1.py
s1 = input("Enter string1: ").lower()
s2 = input("Enter string2: ").lower()
if len(s1) != len(s2):
    print("false")
else:
    for c in s1:
      if s1.count(c) != s2.count(c):
        print("false")
        break
    else:
        print("true")

File: DSA/SAMAGRA_590014891/Day_12_Question_1.c
#include <stdio.h>
#include <string.h>

int isRotation(char *str1, char *str2) {
    if (strlen(str1) != strlen(str2))
        return 0;

    char concat[2000];
    strcpy(concat, str1);
    strcat(concat, str1);

    if (strstr(concat, str2) != NULL)
        return 1;
    else
        return 0;
}

int main() {
    char str1[1000], str2[1000];
    printf("Enter String1: ");
    fgets(str1, sizeof(str1), stdin);
    printf("Enter String2: ");
    fgets(str2, sizeof(str2), stdin);

    str1[strcspn(str1, "\n")] = '\0';
    str2[strcspn(str2, "\n")] = '\0';

    if (isRotation(str1, str2))
        printf("true\n");
    else
        printf("false\n");

    return 0;
}
